massa-models/src/storage.rs

```diff
@@ -0,0 +1,70 @@
+use crate::prehash::Map;
+use crate::{Block, BlockId, Endorsement, EndorsementId, Operation, OperationId};
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+pub struct StoredBlock {
```

Comment:
Can we, rather than having a `StoredBlock` struct that wrap a block with his serialized representation, have something more like a serialization cache in a separate structure, that would be updated when a block is updated!

Reply:
Could you please elaborate on "separate structure" and "updated when a block is updated"?

Note that the point of having one struct wrap both the block and the serialized representation is to have them both covered by the same lock.

Reply:
Yes I was thinking about having an `HashMap<BlockId, SerializedBlock>` but yes a unique structure make more sense behind a lock mechanism you're right!

Reply:
Having a single map allows for better consistency without having to use more locks. I vote for having either a StoredBlock wrapper like was done here or having the Block structure itself carry its serialized representation so we don't bother with multiple structures. As you prefer